core/state_transition.go

```diff
@@ -452,26 +452,26 @@ func (st *StateTransition) distributeTxFees() error {
 		gatewayFeeRecipient = &common.ZeroAddress
 	}
 
-	governanceAddress, err := vm.GetRegisteredAddressWithEvm(params.GovernanceRegistryId, st.evm)
+	governanceAddress, err := vm.GetRegisteredAddress(st.evm, params.GovernanceRegistryId)
 	if err != nil && err != commerrs.ErrSmartContractNotDeployed && err != commerrs.ErrRegistryContractNotDeployed {
 		return err
 	} else if err != nil {
 		log.Trace("Cannot credit gas fee to community fund: refunding fee to sender", "error", err, "fee", baseTxFee)
-		governanceAddress = &common.ZeroAddress
+		governanceAddress = common.ZeroAddress
```

Comment:
Is this line redundant now? If is it implicit that `GetRegisteredAddress` will always return `ZeroAddress` if it returns an error?

Reply:
my rule of thumb is that if `GetRegisteredAddress` returns an error, them make no assumptions on the other value.

Here it seems "easy" to assume it's zeroValue... but somewhere over the `GetRegisteredAddres` we pass and address variable pointer to an internal funtion that unmarshall a result there.. .what if unmarshalling ends unexpectectly mid process..